packages/liquidator/src/liquidator.js

```diff
@@ -118,18 +121,53 @@ class Liquidator {
           // Override must be one of the default logging levels: ['error','warn','info','http','verbose','debug','silly']
           return Object.values(overrides).every(param => Object.keys(this.logger.levels).includes(param));
         }
+      },
+      whaleDefenseFundWei: {
+        // by default make this disabled
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return this.toBN(x).gte(this.toBN("0"));
+        }
+      },
+      defenseActivationPercent: {
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return parseFloat(x) >= 0 && parseFloat(x) <= 100;
+        }
       }
     };
 
     // Validate and set config settings to class state.
-    Object.assign(this, createObjectFromDefaultProps(config, defaultConfig));
+    const configWithDefaults = createObjectFromDefaultProps(config, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    // this takes in config, and emits log events
+    this.liquidationStrategy = LiquidationStrategy(
+      {
+        ...configWithDefaults,
+        empMinSponsorSize: empProps.minSponsorSize,
+        withdrawalLiveness: empProps.liquidationLiveness
+      },
+      this.web3.utils,
+      (type, severity, data = {}) => {
```

Comment:
You should probably define and name this callback to make it more readable, rather than define upon construction